### DISCUSSION TITLE
Ubuntu1804 and Ubuntu2004 are failing at the Tools.Tests.ps1

### DISCUSSION THREAD

**Discussion by Malaw80s** [actions/virtual-environments#5267] [General questions] [answered]
I'm running to this issue where I see the error below:
I'm at this point of the steps:

`Set-Location C:\virtual-environments

Import-Module .\helpers\GenerateResourcesAndImage.ps1

GenerateResourcesAndImage -SubscriptionId <MY SUBSCRIPTION ID> -ResourceGroupName "<My RSG NAME>" -ImageGenerationRepositoryRoot "$pwd" -ImageType Ubuntu2004 -AzureLocation "East US"`

The script will run for a while then I see the error below:

    azure-arm:
    azure-arm: Running tests from '/imagegeneration/tests/Tools.Tests.ps1'
    azure-arm: Describing Bicep
    azure-arm:   [+] Bicep 947ms (919ms|28ms)
    azure-arm: Tests completed in 1.64s
    azure-arm: Tests Passed: 1, Failed: 0, Skipped: 0 NotRun: 95
==> azure-arm: Provisioning with shell script: C:\virtual-environments\images\linux/scripts/installers/aliyun-cli.sh
    azure-arm: Downloading '/tmp' to './tmp'...
    azure-arm: Error — Either HTTP response code for '/tmp' is wrong - '000' or exit code is not 0 - '3'. Waiting 30 seconds before the next attempt, 19 attempts left
    azure-arm: Error — Either HTTP response code for '/tmp' is wrong - '000' or exit code is not 0 - '3'. Waiting 30 seconds before the next attempt, 18 attempts left
    azure-arm: Error — Either HTTP response code for '/tmp' is wrong - '000' or exit code is not 0 - '3'. Waiting 30 seconds before the next attempt, 17 attempts left
    azure-arm: Error — Either HTTP response code for '/tmp' is wrong - '000' or exit code is not 0 - '3'. Waiting 30 seconds before the next attempt, 16 attempts left
    azure-arm: Error — Either HTTP response code for '/tmp' is wrong - '000' or exit code is not 0 - '3'. Waiting 30 seconds before the next attempt, 15 attempts left
    azure-arm: Error — Either HTTP response code for '/tmp' is wrong - '000' or exit code is not 0 - '3'. Waiting 30 seconds before the next attempt, 14 attempts left
    azure-arm: Error — Either HTTP response code for '/tmp' is wrong - '000' or exit code is not 0 - '3'. Waiting 30 seconds before the next attempt, 13 attempts left
    azure-arm: Error — Either HTTP response code for '/tmp' is wrong - '000' or exit code is not 0 - '3'. Waiting 30 seconds before the next attempt, 12 attempts left
    azure-arm: Error — Either HTTP response code for '/tmp' is wrong - '000' or exit code is not 0 - '3'. Waiting 30 seconds before the next attempt, 11 attempts left
    azure-arm: Error — Either HTTP response code for '/tmp' is wrong - '000' or exit code is not 0 - '3'. Waiting 30 seconds before the next attempt, 10 attempts left
    azure-arm: Error — Either HTTP response code for '/tmp' is wrong - '000' or exit code is not 0 - '3'. Waiting 30 seconds before the next attempt, 9 attempts left
    azure-arm: Error — Either HTTP response code for '/tmp' is wrong - '000' or exit code is not 0 - '3'. Waiting 30 seconds before the next attempt, 8 attempts left
    azure-arm: Error — Either HTTP response code for '/tmp' is wrong - '000' or exit code is not 0 - '3'. Waiting 30 seconds before the next attempt, 7 attempts left
    azure-arm: Error — Either HTTP response code for '/tmp' is wrong - '000' or exit code is not 0 - '3'. Waiting 30 seconds before the next attempt, 6 attempts left
    azure-arm: Error — Either HTTP response code for '/tmp' is wrong - '000' or exit code is not 0 - '3'. Waiting 30 seconds before the next attempt, 5 attempts left
    azure-arm: Error — Either HTTP response code for '/tmp' is wrong - '000' or exit code is not 0 - '3'. Waiting 30 seconds before the next attempt, 4 attempts left
    azure-arm: Error — Either HTTP response code for '/tmp' is wrong - '000' or exit code is not 0 - '3'. Waiting 30 seconds before the next attempt, 3 attempts left
    azure-arm: Error — Either HTTP response code for '/tmp' is wrong - '000' or exit code is not 0 - '3'. Waiting 30 seconds before the next attempt, 2 attempts left
    azure-arm: Error — Either HTTP response code for '/tmp' is wrong - '000' or exit code is not 0 - '3'. Waiting 30 seconds before the next attempt, 1 attempts left
    azure-arm: Error — Either HTTP response code for '/tmp' is wrong - '000' or exit code is not 0 - '3'. Waiting 30 seconds before the next attempt, 0 attempts left
    azure-arm: Could not download /tmp
==> azure-arm: Script exited with non-zero exit status: 1.Allowed exit codes are: [0]
==> azure-arm: Step "StepProvision" failed
==> azure-arm: [c] Clean up and exit, [a] abort without cleanup, or [r] retry step (build may fail even if retry succeeds)?

**Comment by mikhailkoliada** [comment, marked as answer]
@Malaw80s Hi, it has been fixed as a part of https://github.com/actions/virtual-environments/pull/5271

**Reply by Malaw80s** [reply]
Thank you
